tests/test_diffraction_objects.py

```diff
@@ -241,6 +241,24 @@ def _test_valid_diffraction_objects(actual_diffraction_object, function, expecte
     return np.allclose(actual_array, expected_array)
 
 
+def test_get_angle_index():
+    test = DiffractionObject()
+    test.angles = np.array([10, 20, 30, 40, 50, 60])
```

Comment:
I guess we would want to look in `on_tth[0]` not self. Angles.  I'm fact do we need self.angles?

Reply:
Yes it looks like angles should be replaced by on_tth[0]

Reply:
Shall we do get_array_index then? User specifies the xtype and target value, and we return the index they're looking for

Reply:
> Shall we do get_array_index then? User specifies the xtype and target value, and we return the index they're looking for

Yes, that sounds about right